pkg/metrics/timer/timer.go

```diff
@@ -0,0 +1,97 @@
+	stageTimes   map[string]time.Duration
+	startTimes   map[string]time.Time
+}
+
+const (
+	// InfrastructureStageName is a constant string value that is used to display the Infrastructure created stage text in the logs.
+	InfrastructureStageName = "Infrastructure"
+	// APIStageName is a constant string value that is used to display the API stage text in the logs.
+	APIStageName = "API"
+	// BootstrapCompleteStageName is a constant string value that is used to display the Bootstrap Complete text in the logs.
+	BootstrapCompleteStageName = "Bootstrap Complete"
+	// BootstrapDestroyStageName is a constant string value that is used to display the Boostrap Destroy Complete text in the logs.
+	BootstrapDestroyStageName = "Bootstrap Destroy"
+	// ClusterOperatorStageName is a constant string value that is used to display the Cluster Operator Complete text in the logs.
+	ClusterOperatorStageName = "Cluster Operator"
+
+	// ConsoleStageName is a constant string value that is used to display the Cluster Operator Complete text in the logs.
+	ConsoleStageName = "Console"
+
+	// TotalTimeElapsed is a constant string value to denote total time elapsed.
+	TotalTimeElapsed = "Total"
+)
+
+var timer = NewTimer()
+
+// StartTimer initiailzes the timer object with the current timestamp information.
+func StartTimer(key string) {
+	timer.StartTimer(key)
+}
+
+// StopTimer records the duration for the current stage sent as the key parameter and stores the information.
+func StopTimer(key string) {
+	timer.StopTimer(key)
+}
+
+// LogSummary prints the summary of all the times collected so far into the INFO section.
+func LogSummary() {
+	timer.LogSummary()
+}
+
+// NewTimer returns a new timer that can be used to track sections and
+func NewTimer() Timer {
+	return Timer{
+		listOfStages: []string{},
+		stageTimes:   make(map[string]time.Duration),
+		startTimes:   make(map[string]time.Time),
+	}
+}
+
+// StartTimer initiailzes the timer object with the current timestamp information.
+func (t *Timer) StartTimer(key string) {
+	t.listOfStages = append(t.listOfStages, key)
+	t.startTimes[key] = time.Now().Round(time.Second)
+}
+
+// StopTimer records the duration for the current stage sent as the key parameter and stores the information.
+func (t *Timer) StopTimer(key string) {
+	if _, found := t.startTimes[key]; !found {
+		return
+	}
+	duration := time.Now().Round(time.Second).Sub(timer.startTimes[key]).Round(time.Second)
+	t.stageTimes[key] = duration
+}
+
+// LogSummary prints the summary of all the times collected so far into the INFO section.
+func (t *Timer) LogSummary() {
+	indent := ""
+	maxLen := 0
+	for _, item := range t.listOfStages {
+		if len(item) > maxLen && item != TotalTimeElapsed {
+			maxLen = len(item)
+		}
+	}
+
+	if maxLen != 0 {
+		logrus.Debugf("Time elapsed for each section")
+	}
+
+	for _, item := range t.listOfStages {
+		if item != TotalTimeElapsed {
+			logrus.Debugf(indent+"%-"+strconv.Itoa(maxLen)+"s  :  %v", item, t.stageTimes[item])
```

Comment:
I think this line could be cleaned up to make it more readable:

- `indent` appears to prepend an empty string, so can we just drop it?
- use a format string like `Sprintf` (in general string concatenation should be avoided)
- using `*` you can pass in a width as an arg to `Sprintf`, see [here](https://stackoverflow.com/questions/42308070/how-to-dynamically-pass-values-to-sprintf-or-printf) and [here](https://golang.org/pkg/fmt/)
- I still think `:` having its own column is not good